scripts/common.sh

```diff
@@ -55,40 +51,87 @@ function resolve_s3_uri() {
 	fi
 }
 
-function download_delphix_s3_debs() {
-	local pkg_directory="$1"
-	local S3_URI="$2"
-	local tmp_directory
-
-	tmp_directory=$(mktemp -d -p "$TOP/build" tmp.s3-debs.XXXXXXXXXX)
-	pushd "$tmp_directory" &>/dev/null
-
-	aws s3 sync --only-show-errors "$S3_URI" .
-	sha256sum -c --strict SHA256SUMS
-
-	mv ./*deb "$pkg_directory/"
-
-	popd &>/dev/null
-	rm -rf "$tmp_directory"
-}
+#
+# Given an S3 URI pointing to combined-packages artifacts, download all of its
+# artifacts to target directory. If a package name is passed as an argument,
+# then only copy the artifacts for that package.
+#
+# When the combine-packages Jenkins job generates artifacts, it does not
+# copy around the artifacts for individual packages. Rather, it creates a
+# COMPONENTS file that has links to each package's artifacts.
+#
+# When appliance-build is ran via Jenkins, the Jenkins job copies the original
+# combined-packages artifacts to a new S3 location, then dereferences the
+# COMPONENTS file and copies all individual package artifacts into a
+# "packages" directory created under that new S3 location. Jenkins then passes
+# that combined-packages URI to live-build.
+#
+# Thus if a "packages" directory is found under the combined-packages S3 URI,
+# we assume that the dereferencing has already been done and so we just sync
+# the whole directory. Otherwise, we must dereference the COMPONENTS file here
+# and fetch the artifacts for each package.
```

Comment:
Why do we have two different situations, one where the "packages" directory will already exist, and another where it wont yet exist? Can we consolidate to a single mechanism?

Reply:
The situation where the "packages" directory doesn't exist is really just for being able to run `appliance-build` manually without Jenkins. Right now the `combine-packages` job itself doesn't generate the packages directory in order to avoid copying all package artifacts every time that job is called (that would also potentially require cloning a DCoA VM each time we run combine-packages). If we decide to change that in the future, we can remove that check.

Reply:
In what case will the "packages" directory exist? and why does it create the directory in that case?

Reply:
We create the "packages" directory in `appliance-build-stage0` when we copy the package artifacts from their original locations into the new "input-artifacts" S3 location. We do that before calling `appliance-build-stage1`

Reply:
ah, ok.. I see.